src/llmcompressor/modifiers/utils/hooks.py

```diff
@@ -23,24 +23,42 @@ class HooksMixin(BaseModel):
 
     Hooks can be applied to modules or parameters
 
-    Lifecycle:
-        - modifier.register_forward_hook(module, hook)
-        - with HooksMixin.disable_hooks(): model.forward()
-        - modifier.remove_hooks()
+    Typical example
+    >>> modifier.register_forward_hook(module, hook)
+    >>> with HooksMixin.disable_hooks():
+            model.forward(...)
+    >>> modifier.remove_hooks()
+
+    Example of activating only a specific subset of hooks
+    >>> hooks = [modifier.register_forward_hook(module, hook) for module in ...]
+    >>> with HooksMixin.disable_hooks(keep=hooks):
+            model.forward(...)
+    >>> modifier.remove_hooks(hooks)
     """
 
-    _HOOKS_DISABLED: ClassVar[bool] = False  # attached to global HooksMixin
-    _hooks: Set[RemovableHandle] = set()  # attached to local subclasses
+    # attached to global HooksMixin class
+    _HOOKS_DISABLED: ClassVar[bool] = False
+    _HOOKS_KEEP_ENABLED: ClassVar[Set[RemovableHandle]] = set()
```

Comment:
I think this is also sensitive to my comment below regarding `keep: Set[RemoveableHandle] = set()`. If multiple `HooksMixin` classes are instantiated, they may share this same set. It's super weird and really silly python behavior

Reply:
As mentioned in the comments, this variable is meant to be attached to the global class (Pydantic controls whether it's a class var or instance var using the `ClassVar` type hint)

This is actually intended behavior, as we want the keep set to apply across modifiers

Reply:
resolving after slack huddle, awesome!